net/tcp/tcp_recvfrom.c

```diff
@@ -191,10 +191,10 @@ static inline uint16_t tcp_newdata(FAR struct net_driver_s *dev,
       if (nsaved < buflen)
         {
           nwarn("WARNING: packet data not fully saved "
-                "(%d/%u/%zu/%u bytes)\n",
+                "(%d/%u/%u/%u bytes)\n",
                 buflen - nsaved,
                 (unsigned int)nsaved,
-                recvlen,
+                (unsigned int)recvlen,
```

Comment:
can we relax the %z flag? the library feature is different from language(compiler) feature. libc is provided by NuttX self, and we actually support many c99 feature.

Reply:
I do not see any explicit way to disable `%z` warning while keeping `-Wformat` on. But I agree that since NuttX supports it's own implementation of `printf` we should relax `%z` check.

Reply:
I have updated the patch. ("warning: ISO C90 does not support the ‘z’ gnu_printf length modifier [-Wformat=]" is not required to fix).